include/arc_utilities/math_helpers.hpp

```diff
@@ -73,10 +81,13 @@ namespace EigenHelpers //TODO: Change namespace to ArcMath, breaking change
         }
     }
 
-    inline double Interpolate(const double p1, const double p2, const double ratio)
+    template <typename FloatType>
+    inline FloatType SafetyCheckRatio(const FloatType ratio)
```

Comment:
I like pulling this out. I don't love the name.
What does it mean to "safety check a ratio"? Why is 1.93 an invalid "ratio"?

I suggest:
`SafetyCheckUnitInterval`

Other alternatives:
`SafetyCheckProportion`
`SafetyCheckPortion`
`SafetyCheckInterpolationRatio`

Reply:
Or:
`inline FloatType SafetyCheckInterval(const FloatType ratio, const FloatType lower_bound=0, const FloatType upper_bound=1)

Reply:
I'm going to go with `SafetyCheckUnitInterval` for now; `arc_helpers.hpp` has a `ClampValueAndWarn` that does the second option already.